Core/include/Acts/Vertexing/AdaptiveMultiVertexFitter.ipp

```diff
@@ -188,25 +181,42 @@ bool Acts::AdaptiveMultiVertexFitter<input_track_t, linearizer_t>::
 }
 
 template <typename input_track_t, typename linearizer_t>
-Acts::Result<void> Acts::
-    AdaptiveMultiVertexFitter<input_track_t, linearizer_t>::prepareVertexForFit(
+Acts::Result<void>
+Acts::AdaptiveMultiVertexFitter<input_track_t, linearizer_t>::
+    updateImpactParams3D(
         State& state, Vertex<input_track_t>* vtx,
         const VertexingOptions<input_track_t>& vertexingOptions) const {
   // The current vertex info object
   auto& currentVtxInfo = state.vtxInfoMap[vtx];
-  // The seed position
-  const Vector3& seedPos = currentVtxInfo.seedPosition.template head<3>();
+  // Vertex position, i.e., point wrt which the impact parameters are estimated
+  const Vector3& vtxPosition = currentVtxInfo.oldPosition.template head<3>();
 
   // Loop over all tracks at current vertex
   for (const auto& trk : currentVtxInfo.trackLinks) {
+    // Boolean indicating whether the impact parameters were already calculated
+    bool hasIP = currentVtxInfo.impactParams3D.count(trk) == 1;
+    // Track parameters
+    auto trkParams = m_extractParameters(*trk);
+    // Origin of the track reference surface
+    Vector3 surfaceOrigin = trkParams.referenceSurface()
+                                .transform(vertexingOptions.geoContext)
+                                .translation();
+    // Skip the impact point estimation if the impact parameters of trk wrt the
+    // vertex position were already calculated
+    if (hasIP and surfaceOrigin == vtxPosition) {
+      continue;
+    }
```

Comment:
It's probably not critical, but `count` does a map lookup which is not free. You could use `&&` in the `if` and have it only perform the lookup if the other condition `surfaceOrigin == vtxPosition` is false, by not pre-computing the `hasIP` bool.

Also, the float equality check is a bit awkward. There's no other way to check this?

Reply:
Done!

Since we set the origin of the track reference surface exactly to the vertex position in `estimate3DImpactParameters` (see screenshot) I thought the float comparison was fine... Is that not the case? 

![image](https://github.com/acts-project/acts/assets/72298366/f45160a0-ee98-44e4-9323-c93c05188dc8)

Reply:
Yes, it should be ~ fine.